docs/src/pages/guides/composition/composition.md

```diff
@@ -127,40 +127,70 @@ Here is a demo with [React Router DOM](https://github.com/ReactTraining/react-ro
 
 You can find the details in the [TypeScript guide](/guides/typescript#usage-of-component-property).
 
-### Caveat with refs
+## Caveat with refs
```

Comment:
Shouldn't this be the same heading level as the other caveats?

```suggestion
### Caveat with refs
```

Reply:
This section describes caveats with refs for the `component` prop as well as for children. Using `###` would imply it belongs to `## Component property`